modules/budgets/app/controllers/budgets_controller.rb

```diff
@@ -136,11 +137,21 @@ def update
   end
 
   def destroy
+    budget = Budget.find(params[:id])
+    if budget.work_packages.any?
+      redirect_to destroy_info_budget_path(budget)
```

Comment:
I'd suggest to do that in a `before_action` to make the signature of destroy smaller. Then you also don't need the `return` statement.